cupy/cuda/memory.pyx

```diff
@@ -361,7 +361,7 @@ cdef class SingleDeviceMemoryPool:
             except runtime.CUDARuntimeError as e:
                 if e.status != runtime.errorMemoryAllocation:
                     raise
-                self.free_all_free()
+                self.free_all_blocks()
```

Comment:
This line is not relevant to performance. Could you remove this change and send another PR to fix the issue?